oceannavigator/frontend/src/components/OceanNavigator.jsx

```diff
@@ -123,6 +123,16 @@ export default class OceanNavigator extends React.Component {
     this.updateScale = this.updateScale.bind(this);
     this.get_variables_promise = this.get_variables_promise.bind(this);
     this.get_timestamp_promise = this.get_timestamp_promise.bind(this);
+    this.setStartTime = this.setStartTime.bind(this);
+    // Setting the starttime and time variable default value
+    const time_promise = this.get_timestamp_promise("giops_day", "votemper");
+      $.when(time_promise).done(function(time) {
+        let newTime = time[time.length-1].id;
+        let newStarttime = time[0].id;
+        this.state.time = newTime;
+        this.state.starttime = newStarttime;
+        this.setState(state);
```

Comment:
What do you think of simplifying to the following? Also, you should avoid "direct state mutation" (i.e. use `this.setState` over `this.state.myVariable = `. This is because React does a bunch of async batched updates under the hood with `setState` to improve performance.
```suggestion
        const newTime = time[time.length-1].id;
        const newStarttime = time[0].id;
        this.setState({time: newTime, starttime: newStarttime});
```

Reply:
Used `this.setState()` instead of the direct state mutation.